src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1016,21 +1038,23 @@ namespace cryptonote
     std::vector<tx_verification_batch_info> tx_info;
     tx_info.reserve(tx_blobs.size());
     for (size_t i = 0; i < tx_blobs.size(); i++) {
-      if (!results[i].res || already_have[i])
+      if (!results[i].valid || already_have[i])
         continue;
-      tx_info.push_back({&results[i].tx, results[i].hash, tvc[i], results[i].res});
+      tx_info.push_back({&results[i].tx, results[i].hash, tvc[i], results[i].valid});
     }
     if (!tx_info.empty())
       handle_incoming_tx_accumulated_batch(tx_info, tx_relay == relay_method::block);
 
+    bool valid_events = false;
     bool ok = true;
     it = tx_blobs.begin();
     for (size_t i = 0; i < tx_blobs.size(); i++, ++it) {
-      if (!results[i].res)
+      if (!results[i].valid)
       {
         ok = false;
         continue;
       }
+      results[i].valid = false;
```

Comment:
Why ? Looks like you set valid to false for txes we alreay have in the txpool or the chain, and therefore are valid.

Reply:
This is probably the result of several edits, and is now confusing. I will remove this line and 1074, then add an `else { results[i].valid = false }` section below the `if` block that starts on line 1071. The intent was to make sure that txes _not_ added to the pool are marked as `!invalid` so that they are not reported over ZMQ (the ZMQ notify code skips those txes).